provider/consul/consul_catalog.go

```diff
@@ -28,11 +29,14 @@ var _ provider.Provider = (*CatalogProvider)(nil)
 
 // CatalogProvider holds configurations of the Consul catalog provider.
 type CatalogProvider struct {
-	provider.BaseProvider `mapstructure:",squash"`
-	Endpoint              string `description:"Consul server endpoint"`
-	Domain                string `description:"Default domain used"`
-	Prefix                string `description:"Prefix used for Consul catalog tags"`
-	client                *api.Client
+	provider.BaseProvider      `mapstructure:",squash"`
+	Endpoint                   string `description:"Consul server endpoint"`
+	Domain                     string `description:"Default domain used"`
+	Prefix                     string `description:"Prefix used for Consul catalog tags"`
+	FrontEndRule               string `description:"Frontend rule used for Consul services"`
+	AllTagsConstraintFiltering bool   `description:"Should use all tags for constraint filtering"`
```

Comment:
The use-case here is to enable the ability to filter (use tag constraints) on the existing Consul tags, not only the ones that are defined in `traefik.tags`  Many places with existing Consul infrastructure already have quite an elaborate tagging setup for their service registrations, so being able to leverage those for filtering is a big plus.  For example, each Spring Boot application that uses `sping-cloud-consul` module, will automatically register the management endpoint using `management` tag.  Because it might not be desired to expose LB routes to management endpoints, one might want to create a constraint rule like `constraints = ["tag!=management"]`, and when coupled with `AllTagsConstraintFiltering=true`, it will use the "raw" `management` tag value, instead of only the values provided by `traefik.tags` tag.  Does this make things clearer?